Automattic/wp-calypso
# Domain Management i1: show indicators when bulk jobs are happening

Related to https://github.com/Automattic/dotcom-forge/issues/3419

## Proposed Changes

It's a work in progress but is starting to come together:

* Support polling for bulk status updates and showing jobs in progress
    - Checks on load and decides whether to keep polling or not
    - Shows spinner on domains with pending updates
    - Shows global status of all running bulk jobs  
    - calls API to delete old jobs when popover is closed. This will clear the top-level notification the next time `/GET domains/bulk-actions` is called if no other jobs are running. 

**The global popup shows domain-level status updates. This is a rabbit hole, I feel.  Maybe we just want to show that a job is in progress and then report any errors if something was wrong?**

**Global level**:

<img width="982" alt="Screenshot 2566-09-05 at 12 06 29" src="https://github.com/Automattic/wp-calypso/assets/6851384/e83ad927-8bb1-4e42-bafb-b4e877fd236d">

![image](https://github.com/Automattic/wp-calypso/assets/6851384/f5c084df-f1ae-4e47-bf42-07afff038f51)


**Domain level:**
![image](https://github.com/Automattic/wp-calypso/assets/6851384/23b03334-398e-4917-b83c-a573536fe66a)



## Testing Instructions



* Apply D120555-code your sandbox, so deleting completed jobs will work
* Checkout this branch
* Navigate to `domains/manage?flags=domains/management`
* Select one or more domains and submit a bulk auto-renew update
* Confirm that you see the indicators above on the domain level and in the header
* Open your browser's Network tools and confirm that `domains/bulk-actions` is being called every six seconds
* Make sure your sandbox is set up for the new async jobs as per "configure sandbox" here: D119397-code 
* Go to your sandbox and run `cat /tmp/php-errors` and get the async command. 
* Run the command, e.g. `async-jobs/tester.php 87169417713 wpj_` 
    -  Watch the domain(s) you updated and the spinner should clear. 
    - At global level, the bulk job should go from orange to green status
    - The API calls to `domains/bulk-actions` should stop 
* For additional tests, you can try modifying endpoint `GET /domains/bulk-actions` in `wp-content/rest-api-plugins/endpoints/domains-bulk-actions.php` to return different payloads, or just modify `packages/data-stores/src/queries/use-bulk-domain-update-status-query.ts` in the client. 


## Pre-merge Checklist
